pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -533,6 +544,11 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 			}
 			switch pluginParams.ListenerProtocol {
 			case plugin.ListenerProtocolHTTP:
+				if noneMode {
+					// TODO: in future we may use the Sidecar config to still bind http ports. This will require DNS interception
+					// or /etc/hosts, to resolve the http requests to 127.0.0.1
+					continue // will use HTTP PROXY
+				}
```

Comment:
I am skeptical of this design choice where TCP services get ports on the local host while all HTTP stuff is packed into a single proxy. This will break all the virtual service configs or force users to explicitly specify the service port (if not specified, in some conditions it defaults to listener port - which in this case is 15001). This gets also complicated when you have a service with HTTP on multiple ports because you will have multiple VirtualHosts with same DNS name. Will result in incorrect routing.

Reply:
On HTTP: it's not a problem, the Host header includes port and routing is fine. 

We do want to also support binding to port for HTTP services - but it's a more intrusive change in the app comparing to just setting a flag or env variable ( virtually all languages and libraries support this). 

For TCP: AFAIK all configs use the service port ( except for containerPort used in ServiceEntry endpoints ). That's the point of this feature - the user manages the ports and each TCP service gets allocated a port.